### PR TITLE
(176) Placement request form

### PR DESCRIPTION
**Pull request by richpjames**
# Context
Building on #807 this adds more pages to the Request a Placement form. 
We also add the submission endpoint and confirmation page. 

[Trello card](https://trello.com/c/ncPgE6LL/176-add-a-flow-to-create-and-submit-a-placement-request)

# Changes in this PR
- Initially some tidying up and corrections from #807. 
- We then add the majority of the form pages. 
- Before adding the 'Check Your Answers' page we have to adapt some of the CYA utils used for Applications/Assessments to be able to accept Placement Applications. We need to add some unique utils to build the Placement Application as the journey only has one one section and task and we want the individual pages to be shown on individual cards on the CYA screen. This is due to the PA flow not having a tasklist so we need to keep all the pages in one task. 
- We then add the Confirmation page 
- We then retroactively add the 'submission' functionality to submit the PA in the API when the user has passed the CYA screen 

## Screenshots of UI changes
check-your-answers
dates-of-placement
previous-rotl
reasons
same-ap
updates-to-application

### CODE REVIEW

File: server/form-pages/placement-application/request-a-placement/checkYourAnswers.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  next() {
+    return ''
+  }
+
+  response() {
+    return {}
+  }
+
+  errors() {
```
Comment (pezholio) [inline server/form-pages/placement-application/request-a-placement/checkYourAnswers.ts:31]: Should we ensure they've checked the box here?

**Review by pezholio — APPROVED** [review]
It's a lot of commits, but I've given it a once over. Looks great though! Just a couple of things to consider 👍

File: server/form-pages/placement-application/request-a-placement/checkYourAnswers.ts (hunk 1 above)
Comment (richpjames) [reply server/form-pages/placement-application/request-a-placement/checkYourAnswers.ts:31]: We could add the validation but it wouldn't be handled here (agree this is confusing) - it's handled in the placementRequestController's submit method

File: integration_tests/pages/match/placementRequestForm/previousRotlPlacement.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export default class PreviousRotlPlacement extends Page {
+  constructor() {
+    super('Request for placement')
+  }
+
+  completeForm() {
+    this.checkRadioByNameAndValue('previousRotlPlacement', 'yes')
+    this.getTextInputByIdAndClear('lastAp')
+    this.getTextInputByIdAndEnterDetails('lastAp', 'Some Manager')
```
[the remark this answers is not in the document]
Comment (richpjames) [reply integration_tests/pages/match/placementRequestForm/previousRotlPlacement.ts:11]: Good plan

File: server/form-pages/placement-application/request-a-placement/checkYourAnswers.ts (hunk 1 above)
Comment (pezholio) [reply server/form-pages/placement-application/request-a-placement/checkYourAnswers.ts:31]: D'oh, course! 🤦

File: server/utils/placementRequests/placementApplicationSubmissionData.test.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { addResponseToApplication, addResponsesToApplication } from '../../testutils/addToApplication'
+import { placementApplicationFactory } from '../../testutils/factories'
+import { placementApplicationSubmissionData } from './placementApplicationSubmissionData'
+
+jest.mock('../../form-pages/utils')
+
+describe('placementApplicationSubmissionData', () => {
+  it('returns the data in the correct format for submission', () => {
+    let placementApplication = placementApplicationFactory.build()
+    placementApplication = addResponseToApplication(placementApplication, {
```
[the remark this answers is not in the document]
Comment (richpjames) [reply server/utils/placementRequests/placementApplicationSubmissionData.test.ts:12]: Yep

**Pull request merged**